### ISSUE TITLE
2.1.0 build failure on FreeBSD

### ISSUE THREAD

**Issue by Jehops** [bbuchfink/diamond#652]
Hello,

Diamond 2.1.0 fails to build on FreeBSD 13.1 using clang 13.0.0.

```
FAILED: CMakeFiles/arch_generic.dir/src/dp/score_profile.cpp.o 
/usr/local/libexec/ccache/c++ -DLEFTMOST_SEED_FILTER -DMAX_SHAPE_LEN=19 -DSEQ_MASK -DSTRICT_BAND -I/wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/lib -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -Wall -Wextra -Wno-implicit-fallthrough -Wreturn-type -Wno-unused -Wno-unused-parameter -Wno-unused-variable -Wno-uninitialized -Wno-deprecated-copy -Wno-unknown-warning-option -Wno-pragma-clang-attribute -Wno-overloaded-virtual -Wno-missing-braces -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -DDISPATCH_ARCH=ARCH_GENERIC -DARCH_ID=0 -DEigen=Eigen_GENERIC -std=gnu++14 -MD -MT CMakeFiles/arch_generic.dir/src/dp/score_profile.cpp.o -MF CMakeFiles/arch_generic.dir/src/dp/score_profile.cpp.o.d -o CMakeFiles/arch_generic.dir/src/dp/score_profile.cpp.o -c /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/score_profile.cpp
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/score_profile.cpp:1:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/score_profile.h:24:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../basic/sequence.h:29:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../basic/../util/binary_buffer.h:25:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../basic/../util/algo/varint.h:26:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../basic/../util/algo/../system/endianness.h:5:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../basic/../util/algo/../system/../../lib/psnip/endian.h:33:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../basic/../util/algo/../system/../../lib/psnip/builtin.h:60:
In file included from /usr/lib/clang/13.0.0/include/x86intrin.h:15:
In file included from /usr/lib/clang/13.0.0/include/immintrin.h:62:
/usr/lib/clang/13.0.0/include/avxintrin.h:4977:25: error: '__target__' attribute only applies to functions
static __inline __m256i __DEFAULT_FN_ATTRS
```
Full build log: http://pkg.ftfl.ca/data/13amd64-default/2023-01-30_15h17m08s/logs/errors/diamond-2.1.0.log

It also fails on FreeBSD 14 (unreleased) using clang 14.0.5.
```
FAILED: CMakeFiles/arch_generic.dir/src/dp/ungapped_simd.cpp.o 
/usr/local/libexec/ccache/c++ -DLEFTMOST_SEED_FILTER -DMAX_SHAPE_LEN=19 -DSEQ_MASK -DSTRICT_BAND -I/wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/lib -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -Wall -Wextra -Wno-implicit-fallthrough -Wreturn-type -Wno-unused -Wno-unused-parameter -Wno-unused-variable -Wno-uninitialized -Wno-deprecated-copy -Wno-unknown-warning-option -Wno-pragma-clang-attribute -Wno-overloaded-virtual -Wno-missing-braces -O2 -pipe -fstack-protector-strong -fno-strict-aliasing   -DNDEBUG -DDISPATCH_ARCH=ARCH_GENERIC -DARCH_ID=0 -DEigen=Eigen_GENERIC -std=gnu++14 -MD -MT CMakeFiles/arch_generic.dir/src/dp/ungapped_simd.cpp.o -MF CMakeFiles/arch_generic.dir/src/dp/ungapped_simd.cpp.o.d -o CMakeFiles/arch_generic.dir/src/dp/ungapped_simd.cpp.o -c /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/ungapped_simd.cpp
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/ungapped_simd.cpp:25:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/ungapped.h:23:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../util/geo/diagonal_segment.h:24:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../util/geo/../../basic/sequence.h:29:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../stats/../basic/../util/binary_buffer.h:25:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../util/algo/varint.h:26:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../util/algo/../system/endianness.h:5:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../util/algo/../system/../../lib/psnip/endian.h:33:
In file included from /wrkdirs/usr/ports/biology/diamond/work/diamond-2.1.0/src/dp/../util/algo/../system/../../lib/psnip/builtin.h:60:
In file included from /usr/lib/clang/14.0.5/include/x86intrin.h:15:
In file included from /usr/lib/clang/14.0.5/include/immintrin.h:66:
/usr/lib/clang/14.0.5/include/avxintrin.h:4808:1: error: expected ')'
_mm256_set_m128i (__m128i __hi, __m128i __lo)
^
```
Full build log: http://pkg.ftfl.ca/data/14amd64-default/2023-01-30_15h37m48s/logs/errors/diamond-2.1.0.log

If there is anything other information that I can provide or anything to test, please let me know.

**Comment by bbuchfink** [comment]
Sorry about this, I will post an updated release in the next days.

**Comment by bbuchfink** [comment]
This should be fixed in version 2.1.1.

**Comment by Jehops** [comment]
I confirm that 2.1.1 builds.  Thank you.
